frontend/src/scenes/notebooks/Notebook/Notebook.tsx

```diff
@@ -50,173 +34,75 @@ export function Notebook({ shortId, editable = false }: NotebookProps): JSX.Elem
-            // LinkExtension.configure({}),
-        ],
-        content,
-        editorProps: {
-            attributes: {
-                class: 'NotebookEditor',
-            },
-            handleDrop: (view, event, _slice, moved) => {
-                const editor = logic.values.editor // Only for type checking - should never be null
-                if (!editor) {
-                    return false
-                }
-
-                if (!moved && event.dataTransfer) {
-                    const text = event.dataTransfer.getData('text/plain')
-                    const node = event.dataTransfer.getData('node')
-                    const properties = event.dataTransfer.getData('properties')
-
-                    if (text.indexOf(window.location.origin) === 0 || node) {
-                        // PostHog link - ensure this gets input as a proper link
-                        const coordinates = view.posAtCoords({ left: event.clientX, top: event.clientY })
-
-                        if (!coordinates) {
-                            return false
-                        }
-
-                        if (node) {
-                            editor
-                                .chain()
-                                .focus()
-                                .setTextSelection(coordinates.pos)
-                                .insertContent({ type: node, attrs: JSON.parse(properties) })
-                                .run()
-
-                            // We report this case, the pasted version is handled by the posthogNodePasteRule
-                            posthog.capture('notebook node dropped', { node_type: node })
-                        } else {
-                            editor?.chain().focus().setTextSelection(coordinates.pos).run()
-                            view.pasteText(text)
-                        }
-
-                        return true
-                    }
-
-                    if (event.dataTransfer.files && event.dataTransfer.files[0]) {
-                        // if dropping external files
-                        const file = event.dataTransfer.files[0] // the dropped file
-
-                        console.log('TODO: Dropped file!', file)
-                        // TODO: Detect if it is an image and add image upload handler
-
-                        return true
-                    }
-                }
-
-                return false
-            },
-        },
-        onUpdate: ({}) => {
-            if (ignoreUpdateRef.current) {
-                ignoreUpdateRef.current = false
-                return
-            }
-            onEditorUpdate()
-        },
-    })
-
     useEffect(() => {
-        if (_editor) {
-            setEditorRef(_editor)
+        if (editor) {
+            editor.setEditable(editable)
         }
-    }, [_editor])
-
-    useEffect(() => {
-        _editor?.setEditable(editable && !!notebook, false)
-    }, [editable, _editor, notebook])
+    }, [editor, editable])
```

Comment:
I guess we could pass `editable` in as a prop to the logic and make this call in a listener to `setEditor` but not blocking